### PR TITLE
Remove Dummy Declarations for GCC

### PR DESCRIPTION
**Pull request by gzm0**
This makes GCC less special in the Emitter setup.

### CODE REVIEW

File: linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/KnowledgeGuardian.scala [hunk 1]
```diff
@@ -369,6 +360,88 @@ private[emitter] final class KnowledgeGuardian(config: CommonPhaseConfig) {
...
+    }
+
+    /** Call this when we invalidate all caches. */
+    def unregisterAll(): Unit = {
+      methodsInHijackedClassesAskers.clear()
+    }
+  }
+
+  private def invalidateAskers(askers: mutable.Set[Invalidatable]): Unit = {
+    /* Calling `invalidateAndUnregisterFromAll()` will cause the
```
Comment (sjrd) [inline linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/KnowledgeGuardian.scala]: There seems to be a mismatch between `invalidateAndUnregisterFromAll` in the comment versus `invalidate` in the actual code. I know it was already like that before the change, but maybe use the opportunity to fix the inconsistency?

Comment (gzm0) [reply linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/KnowledgeGuardian.scala]: Good catch.

**Comment by gzm0** [comment]
Updated.

**Comment by gzm0** [comment]
Updated Mima config.

**Pull request merged**